typescript/utils/image_utils.ts

```diff
@@ -29,7 +26,8 @@ import {argbFromRgb} from './color_utils.js';
 export async function sourceColorFromImage(image: HTMLImageElement) {
   // Convert Image data to Pixel Array
   const imageBytes = await new Promise<Uint8ClampedArray>((resolve, reject) => {
-    const canvas = document.createElement('canvas');
+    const element = document.createElement('canvas');
+    const canvas = 'OffscreenCanvas' in window ? element.transferControlToOffscreen() : element;
```

Comment:
Does it make any sense to use `transferControlToOffscreen()` without using the web worker? 🤔

If not, does it make any sense to move the `callback` function code into the web worker? 🤔